File: XMLCheck/Handler/CheckRulesXML05/Xml5RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML05
{
    public abstract class Xml5RuleBase:IXml5Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml5Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml5/xml5RuleBase.ts
import { Xml5Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';

export interface IXml5Rule {
  readonly key: string;
  check(model: Xml5Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml5RuleBase implements IXml5Rule {
  public abstract get key(): string;
  public abstract check(model: Xml5Model, context: HoSoContext): ErrorDetails | null;

  protected error(message: string): ErrorDetails {
    return {
      key: this.key,
      message,
    };
  }
}
